presto-tpch/src/main/java/com/facebook/presto/tpch/TpchRecordSet.java

```diff
@@ -26,42 +31,61 @@
 
 import java.util.Iterator;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
 
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
 import static com.facebook.presto.tpch.TpchMetadata.getPrestoType;
 import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Iterables.transform;
+import static java.lang.String.format;
 import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
 
 public class TpchRecordSet<E extends TpchEntity>
         implements RecordSet
 {
     public static <E extends TpchEntity> TpchRecordSet<E> createTpchRecordSet(TpchTable<E> table, double scaleFactor)
     {
-        return createTpchRecordSet(table, table.getColumns(), scaleFactor, 1, 1);
+        return createTpchRecordSet(table, table.getColumns(), scaleFactor, 1, 1, Optional.empty());
     }
 
     public static <E extends TpchEntity> TpchRecordSet<E> createTpchRecordSet(
             TpchTable<E> table,
             Iterable<TpchColumn<E>> columns,
             double scaleFactor,
             int part,
-            int partCount)
+            int partCount,
+            Optional<TupleDomain<ColumnHandle>> predicate)
     {
-        return new TpchRecordSet<>(table.createGenerator(scaleFactor, part, partCount), columns);
+        return new TpchRecordSet<>(table.createGenerator(scaleFactor, part, partCount), columns, predicate);
     }
 
     private final Iterable<E> table;
     private final List<TpchColumn<E>> columns;
     private final List<Type> columnTypes;
+    private final List<TpchColumnHandle> columnHandles;
+    private final Optional<Predicate<Map<ColumnHandle, NullableValue>>> predicate;
 
-    public TpchRecordSet(Iterable<E> table, Iterable<TpchColumn<E>> columns)
+    public TpchRecordSet(Iterable<E> table, Iterable<TpchColumn<E>> columns, Optional<TupleDomain<ColumnHandle>> predicate)
```

Comment:
It's called predicate in similar context in code base (e.g: ConnectorTableLayout), so I am leaving it as is.